coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java

```diff
@@ -89,6 +106,16 @@ private static String constructSparkSQL(RelNode sparkRelNode) {
     return rewritten.toSqlString(SparkSqlDialect.INSTANCE).getSql();
   }
 
+  private static String constructSparkSQLWithExplicitAlias(RelNode sparkRelNode, List<String> aliases) {
+    SparkRelToSparkSqlConverter rel2sql = new SparkRelToSparkSqlConverter();
+    // Create temporary objects r and rewritten to make debugging easier
+    SqlImplementor.Result r = rel2sql.visitChild(0, sparkRelNode);
+    SqlNode rewritten = r.asStatement().accept(new SparkSqlRewriter());
+    // Use a second pass visit to add explicit alias names
+    SqlNode aliasAdded = rewritten.accept(new AddExplicitAlias(aliases));
```

Comment:
Can we embed the alias addition in `SparkSqlRewriter` as we did originally?

Reply:
`SparkSqlRewriter` also overrides `public SqlNode visit(SqlCall call)` and I feel tangling the logic there is not very clean, as the transformation there (manipulating the `CAST` might interfere with the alias manipulation) Also I only want this visitor to be applied with the new API code path.